src/transactions.ts

```diff
@@ -151,15 +158,21 @@ export class SubscriptionTransactionWrapper implements ReadTransaction {
   scan<Options extends ScanOptions, Key extends KeyTypeForScanOptions<Options>>(
     options?: Options,
   ): ScanResult<Key, ReadonlyJSONValue> {
-    this._scans.push(toDbScanOptions(options));
-    return this._tx.scan(options);
+    const scanInfo: ScanSubrscriptionInfo = {
+      options: toDbScanOptions(options),
+    };
+    this._scans.push(scanInfo);
+    return this._tx.scan(options, (key, isLastBeforeLimit, isLastEntry) => {
+      scanInfo.endInclusiveKey =
+        isLastEntry && !isLastBeforeLimit ? undefined : key;
```

Comment:
This does not look right. See other comment

Reply:
I've removed this as it was only needed for the version that tried to optimize the case where the scan was not read to limit.